sql/hive/src/test/scala/org/apache/spark/sql/hive/ParquetHiveCompatibilitySuite.scala

```diff
@@ -137,8 +141,190 @@ class ParquetHiveCompatibilitySuite extends ParquetCompatibilityTest with TestHi
   }
 
   test("SPARK-16344: array of struct with a single field named 'array_element'") {
+
     testParquetHiveCompatibility(
       Row(Seq(Row(1))),
       "ARRAY<STRUCT<array_element: INT>>")
   }
+
+  test(s"SPARK-12297: Parquet Timestamp & Hive timezone") {
```

Comment:
I think it would be better to have separate test cases for adjustments when reading, adjustments when writing and setting the table property when creating tables.